### DISCUSSION TITLE
Missing vtk header file

### DISCUSSION THREAD

**Discussion by Kgeorge1729** [Chaste/Chaste#150] [Q&A] [answered]
When running the test TestVisualizingWithParaviewTutorial, I keep running into the error vtkDataArray.h: No such file or directory. To try and fix this, I've tried rebuilding Chaste with ccmake. I noticed that the variable VTK_DIR was set to /usr/local/lib/cmake/vtk-890. However, vtkDataArray.h is in a different directory. When I tried to change this variable value to the directory containing the header, ccmake changes this value back when I try to configure the new make file. Does anyone have any insight on why this might be the case?

**Comment by fcooper8472** [comment]
From your build directory (assuming it's a subdirectory of the main Chaste directory), can you run

```
cmake ..
```

and copy the entire output into this thread?

**Reply by Kgeorge1729** [reply]
-- CMake version: 3.16.3
-- Operating system detected as...
-- 	... Ubuntu 20.04 (focal): Linux-5.4.0-155-generic
-- Adding build types...
-- Current build type is : Debug
-- Adding compiler flags...
-- 	...for GNU compiler, version 9.4.0
-- Updating pip setuptools and wheel
Requirement already satisfied: pip in ./codegen_python3_venv/lib/python3.8/site-packages (23.2.1)
Requirement already satisfied: setuptools in ./codegen_python3_venv/lib/python3.8/site-packages (68.0.0)
Requirement already satisfied: wheel in ./codegen_python3_venv/lib/python3.8/site-packages (0.41.1)
-- Installing chaste_codegen
Requirement already satisfied: chaste_codegen<0.11.0,>=0.10.0 in ./codegen_python3_venv/lib/python3.8/site-packages (from -r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (0.10.2)
Requirement already satisfied: py>=1.10.0 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (1.11.0)
Requirement already satisfied: decorator<5,>=4.4.2 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (4.4.2)
Requirement already satisfied: importlib-metadata<5,>=1.7 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (4.13.0)
Requirement already satisfied: isodate<1,>=0.6.0 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (0.6.1)
Requirement already satisfied: lxml<5,>=4.6.5 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (4.9.3)
Requirement already satisfied: MarkupSafe<2,>=1.1.1 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (1.1.1)
Requirement already satisfied: mpmath<2,>=1.1.0 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (1.3.0)
Requirement already satisfied: networkx<3,>=2.4 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (2.8.8)
Requirement already satisfied: packaging<21,>=20.4 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (20.9)
Requirement already satisfied: Pint<0.20,>=0.9 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (0.19.2)
Requirement already satisfied: pyparsing<3,>=2.4.7 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (2.4.7)
Requirement already satisfied: rdflib<7,>=5.0.0 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (6.3.2)
Requirement already satisfied: six<2,>=1.15.0 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (1.16.0)
Requirement already satisfied: sympy<1.11,>=1.9 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (1.10.1)
Requirement already satisfied: zipp<2,>=1.2.0 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (1.2.0)
Requirement already satisfied: Jinja2<3,>=2.11.3 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (2.11.3)
Requirement already satisfied: cellmlmanip<0.4,>=0.3.6 in ./codegen_python3_venv/lib/python3.8/site-packages (from chaste_codegen<0.11.0,>=0.10.0->-r /home/george/Documents/Programming/Chaste-work/Chaste/chaste_codegen.txt (line 1)) (0.3.6)
-- VTK version: 8.90
-- Determined PETSc install from the pair PETSC_DIR=/usr/lib/petscdir/petsc3.12 PETSC_ARCH=x86_64-linux-gnu-real
-- No HDF5 found in /usr/lib/petscdir/petsc3.12/x86_64-linux-gnu-real: looking for alternative HDF5 instead
-- HDF5: Using hdf5 compiler wrapper to determine C configuration
-- Found MPIEXEC_EXECUTABLE: /usr/bin/mpiexec
-- Found MPIEXEC: /usr/bin/mpiexec
-- Found MPI_CXX_HEADER_DIR: /usr/lib/x86_64-linux-gnu/openmpi/include
-- Found MPI_C_COMPILER: 
-- No ParMETIS found in /usr/lib/petscdir/petsc3.12/x86_64-linux-gnu-real: looking for system ParMETIS instead
-- A library with LAPACK API found.
-- Configuring done
-- Generating done
-- Build files have been written to: /home/george/Documents/Programming/Chaste-work/Chaste-build

here is the output for cmake

**Comment by fcooper8472** [comment]
This is really strange - VTK version 8.90 doesn't seem to exist (at least, it's not a tag on their repository). It looks as though it may be a non-official version that ships with ParaView(?), but either way, it seems that CMake is finding that version, rather than the system version that would have been installed with the chaste-dependencies package.

I've not come across this before.

Can you try, from a fresh build directory:

```
cmake .. -DVTK_DIR=/path/to/vtk/lib/cmake/vtk-X.Y
```

**Reply by Kgeorge1729** [reply, marked as answer]
Did not realize that wasn't an official build. Lord knows where that came from. I've now managed to fix the issue by installing an official version of VTK and regenerating Chaste using cmake. I can now make and run the test I was looking into.
